src/DependencyInjection/Configuration.php

```diff
@@ -148,4 +149,21 @@ private function createMockNode()
 
         return $node;
     }
+
+    private function createToleranceNode()
+    {
+        $treeBuilder = new TreeBuilder();
+        $node = $treeBuilder->root('tolerance');
+
+        $node
+            ->canBeEnabled()
+            ->children()
+                ->scalarNode('retry')->defaultValue(2)->end()
```

Comment:
One question: does it make sense to have a `retry` option, if you can override the `waiter_factory`? For me, this is an implementation-specific configuration value, just like the error voter.

My take on this is that we should simplify this configuration, and just let the user provide a runner, which would be a service configured by the user. Even better, we could make it so that the service could be overridden at the client level, using default options, which can be used in the middleware.

Guzzle middleware can actually handle options, which are either passed directly in the middleware through the client's default options, or through per-request options (that's the means of the `$options` parameter, which is passed to the middleware).